### ISSUE TITLE
rename project to CheckStyle-IDEA

### ISSUE THREAD

**Issue by ravn** [jshiell/checkstyle-idea#194]
On systems with case sensitive file systems the different casing of "checkstyle-idea" and "CheckStyle-IDEA" mean that the default process of cloning from git hub will result in a project where the enclosed IDEA files are not recognized correctly and the project will not build.

Would it be possible to either rename the project files or the github project?

**Comment by jshiell** [comment]
Yeah, I'll rename the project files when I'm at home. Sorry about that.

**Comment by ravn** [comment]
I tried doing so manually (as the plugin link inside IntelliJ times out) but I still have a bit of problems as I am unfamiliar with plugin development in gradle in IntelliJ.  Could you perhaps add the steps needed to build on a newly installed machine (with Java 8 and IntelliJ) with a fresh clone of the sources?

**Comment by jshiell** [comment]
Whoops, that really should be documented.

The TL;DR is 
- add IDEA as an IDEA SDK in the SDKs section of project setup (i.e. where you'd add JDKs).
- open the project

To build, install Gradle (I should probably add the wrapper) and modify the build properties as required, then:

    grade clean installPlugin

I'll do a better write-up anon.

**Comment by jshiell** [comment]
Hmm. Colour me confused. I don't want to change the project name case, as it'll potentially lead to multiple installations of the plugin for people with case-sensitive FSs, but in any case the naming in the project files appears to be correct (i.e. the case matches).

I'll give this a try on a Linux VM when I'm at work and see if I can reproduce the problem.

**Comment by jshiell** [comment]
I've also added some dev setup instructions to `README.md` - feedback would be much appreciated!

**Comment by jshiell** [comment]
I can't reproduce this. I've just cloned it on Ubuntu 15.10 and it's opening in IDEA fine. The only things that appear to need correcting are the SDK and the library paths, as IDEA seems unable to manage downloading them from Maven again.

**Comment by ravn** [comment]
I may be too unfamiliar with the IntelliJ mindset to interpret what I see correctly.  I opened the project, and IntelliJ had _two_ items, one named Checkstyle-IDEA and one named checkstyle-idea (I suppose from the project name).  Hence my theory that the Checkstyle-IDEA was not properly "caught" because of incorrect casing compared to the project name due to the different casing in the file system.

I have now cloned the project a fresh, told IntelliJ to open it, treat it as a Gradle project and it appears to import correctly.  It may have picked up my previous setting of the SDK so I did not have to fix that.  

It appears that IntelliJ is smarter than me, and that things work.  My bad.
